android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2012,11 +2023,10 @@ public static int getKeyboardHeight(Context context) {
     int defaultHeight = (int) context.getResources().getDimension(R.dimen.keyboard_height);
     SharedPreferences prefs = context.getSharedPreferences(context.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
 
-    Display display = ((WindowManager) context.getSystemService(Context.WINDOW_SERVICE)).getDefaultDisplay();
-    int rotation = display.getRotation();
-    if (rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_180) {
+    int orientation = getOrientation(context);
+    if (orientation == Configuration.ORIENTATION_PORTRAIT) {
       return prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, defaultHeight);
-    } else if (rotation == Surface.ROTATION_90 || rotation == Surface.ROTATION_270) {
+    } else if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
```

Comment:
What do we do if ORIENTATION_UNDEFINED is returned? Seems like we fall into a hole?

Reply:
The `defaultHeight` initializes in l. 2023 so that value gets returned in l. 2033. I think typically that will be the portrait dimensions